### ISSUE TITLE
Allow negative values for `-O open_lines`

### ISSUE THREAD

**Issue by arcnmx** [I60R/page#19]
An "empty" value for `-O` implies terminal height, however this does not take the user's prompt height into account. Most users presumably do not want the top X lines of input to scroll away with their prompt. Someone might even like to add an extra line so that the original command prompt that spawned the pager is visible too! Allowing a negative value for `-O` seems like a natural way to augment the auto-detected height, though a separate flag could work too of course.

It also feels like the count is off by one or two; with a terminal 34 lines tall for example...
- `seq 1 33 | page -O` spawns nvim despite it only printing 33 lines without the pager (which is enough room for the prompt too!)
- `seq 1 33 | page -O 34` is equivalent to the above
- `seq 1 33 | page -O 33` spawns nvim despite the input only consisting of exactly 33 lines. This one might be intentional if you assume a prompt is one line long, but it feels wrong?

**Comment by I60R** [comment]
So, do you mean `page -O -2` will imply terminal height minus two lines? That's a very good suggestion!

>It also feels like the count is off by one or two

It's off by one because prompt scrolls away the first line (if prompt is one line, of course) and that makes it very easy to miss a potentially important information. Probably I should make it off by three (!) to handle this properly with popular multiline prompt themes like [powerlevel10k](https://github.com/romkatv/powerlevel10k) and [spaceship](https://github.com/spaceship-prompt/spaceship-prompt) while also giving a good enough defaults for smaller prompts. This behavior should be documented in --help output and users should be able to override it with `-O -value`.

What do you think?

**Comment by arcnmx** [comment]
That's great, thanks!

It still seems off by one though, for example:

```bash
seq 1 $((LINES-3)) | page -O -2
<does not spawn nvim>

seq 1 $((LINES-2)) | page -O -2
<spawns nvim>
```

The former example leaves an extra line at the top, while the latter executes vim despite there being exactly enough room for both the output and prompt. I'd guess it might have to do with trailing newlines causing a miscount?

> Probably I should make it off by three (!) to handle this properly with popular multiline prompt themes

That's up to you! starship's default is 2 lines (which also happens to be what I've used for ages), and I'm not sure whether it's a common need or not to be able to see the bottom line of the prompt that spawned `page`. I'll personally be setting it to two but I'm happy with explicitly doing so.

**Comment by I60R** [comment]
That off by one was intentional to not put "minus one" etc. in documentation but it doesn't make sense anymore.

> EDIT: ahh I see how they use 3, there's an extra blank line before the prompt... obnoxious imo. So yeah, the default ought to be either 1 or 3.

Yeah, 2 would be a perfect value — shows only next prompt for 2-line prompt users and shows either prompts for 1-line prompt users. But 3 is the safest and less surprising one

**Comment by arcnmx** [comment]
daacc907338155b1d7d3be324e2ea684cf63aac9 is looking good, thanks again!

**Comment by arcnmx** [comment]
I've been trying out the latest release, and negative values aren't working for me. For example, `echo hi | page -O -1` spawns nvim while `echo hi | page -O 1` does not. I'm currently testing over SSH, is it just me? I'll test more when I can, but wanted to check whether the feature is currently working for others or not.

**Comment by I60R** [comment]
@arcnmx thank you for reporting!

This was silly mistake: I've overlooked `as usize` behavior.

Seems to be fixed in v3.1.2
